### PR TITLE
d3m job to either trigger or revert

### PR DESCRIPTION
**Pull request by oldchili**
This PR fixes a bug whereby calling `workable()` on the D3M job could cause a state change (in `hub.exec(ilk)`) that was not reverted when `shouldTrigger` returned `false`. As a result, if the D3M job is not the very last job in the storage set `jobs` of the `Sequencer`, any subsequent job could see an invalid state when [`job.workable()` is called](https://github.com/makerdao/dss-cron/blob/22e793e5b03c851e0b8d42c0400c511d3f77afc9/src/Sequencer.sol#L180). This could cause that subsequent job to appear workable (e.g. due to the surplus buffer appearing larger than it actually is) when it should in fact be non-workable.

To fix the issue, this PR makes sure that no state change occurs when the D3M job reports being non-workable.

### CODE REVIEW

**Review by sunbreak1211 — APPROVED** [review]
Looks good

**Review by telome — APPROVED** [review]
LGTM

**Review by telome — APPROVED** [review]

**Review by sunbreak1211 — APPROVED** [review]
Deployed contract looks good

**Review by hexonaut — APPROVED** [review]
LGTM

**Comment by SidestreamColdMelon** [comment]
To be explicit about the state of this PR: is it only waiting for the spell cast, before being merged?

**Comment by sunbreak1211** [comment]
> To be explicit about the state of this PR: is it only waiting for the spell cast, before being merged?

Yes

**Pull request merged**
